.github/workflows/awesome_workflow.yml

```
@@ -1,51 +1,30 @@
 name: Awesome CI Workflow
-
 on: [push, pull_request]
-#   push:
-#     branches: [ master ]
-#   pull_request:
-#     branches: [ master ]
+permissions:
+  contents: write
 
 jobs:
   MainSequence:
     name: Code Formatter
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v1 # v2 is broken for git diff
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v3
```

You need the `fetch-depth` here otherwise it would fail...

```suggestion
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
```